ftl/common/cache_runner.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import httplib2
+import json
+import logging
+import requests
+import subprocess
+import sys
+
+from containerregistry.client import docker_name
+from containerregistry.client import docker_creds
+from containerregistry.client.v2_2 import docker_image
+from containerregistry.client.v2_2 import docker_session
+from containerregistry.transport import transport_pool
+
+from ftl.common import ftl_util
+from ftl.common import constants
+
+# we try/catch these imports so we can reuse this code across
+# all cache images, while basing them off individual language
+# builder images
+try:
+    from ftl.php import layer_builder as php_builder
```

Comment:
Why would this import not work in some of the cases? I'm not sure I understand

Reply:
Yeah I think the master build "package" is the way to go. It should be a matter of adding all the deps to the build rule.

Reply:
I don't think it's that easy unfortunately, because each builder image is based off of a runtime-specific base image that has all the necessary build tools installed in it already (e.g. the python image already has `python`, `pip`, `virtualenv`, etc. configured properly). Adding in the `php` deps to that image would make the import work, but trying to install a `composer.json` package would fail.

The only way I can think to do this (short of building my own base image) is to find a way to "append" the two images together. In theory I think this is possible by unpacking all the layers of each image into a flattened image, but I've never done it successfully. WDYT?